src/traffic_cop/traffic_cop.cpp

```diff
@@ -391,7 +391,17 @@ TrafficCopResult TrafficCop::RunExecutableQuery(const common::ManagedPointer<net
 
   const auto exec_query = portal->GetStatement()->GetExecutableQuery();
 
-  exec_query->Run(common::ManagedPointer(exec_ctx), execution_mode_);
+  try {
+    exec_query->Run(common::ManagedPointer(exec_ctx), execution_mode_);
+  } catch (ExecutionException &e) {
+    /*
+     * An ExecutionException is thrown in the case of some failure caused by a software bug or caused by some data
+     * exception. In either case we abort the current transaction and return an error to the client.
+     */
+    connection_ctx->Transaction()->SetMustAbort();
```

Comment:
Let's add a TODO that in the future we might want to evict this plan from the statement cache (no functionality for that at the moment). No sense in keeping a busted plan around.

Reply:
Done...hope you don't mind I assigned it to you.